beetsplug/fetchart.py

```diff
@@ -39,191 +38,166 @@
 CONTENT_TYPES = ('image/jpeg',)
 DOWNLOAD_EXTENSION = '.jpg'
 
-log = logging.getLogger('beets')
-
 requests_session = requests.Session()
 requests_session.headers = {'User-Agent': 'beets'}
 
 
-def _fetch_image(url):
-    """Downloads an image from a URL and checks whether it seems to
-    actually be an image. If so, returns a path to the downloaded image.
-    Otherwise, returns None.
-    """
-    log.debug(u'fetchart: downloading art: {0}', url)
-    try:
-        with closing(requests_session.get(url, stream=True)) as resp:
-            if 'Content-Type' not in resp.headers \
-                    or resp.headers['Content-Type'] not in CONTENT_TYPES:
-                log.debug(u'fetchart: not an image')
-                return
-
-            # Generate a temporary file with the correct extension.
-            with NamedTemporaryFile(suffix=DOWNLOAD_EXTENSION, delete=False) \
-                    as fh:
-                for chunk in resp.iter_content():
-                    fh.write(chunk)
-            log.debug(u'fetchart: downloaded art to: {0}',
-                      util.displayable_path(fh.name))
-            return fh.name
-    except (IOError, requests.RequestException):
-        log.debug(u'fetchart: error fetching art')
-
-
 # ART SOURCES ################################################################
 
-# Cover Art Archive.
-
-CAA_URL = 'http://coverartarchive.org/release/{mbid}/front-500.jpg'
-CAA_GROUP_URL = 'http://coverartarchive.org/release-group/{mbid}/front-500.jpg'
-
-
-def caa_art(album):
-    """Return the Cover Art Archive and Cover Art Archive release group URLs
-    using album MusicBrainz release ID and release group ID.
-    """
-    if album.mb_albumid:
-        yield CAA_URL.format(mbid=album.mb_albumid)
-    if album.mb_releasegroupid:
-        yield CAA_GROUP_URL.format(mbid=album.mb_releasegroupid)
-
-
-# Art from Amazon.
-
-AMAZON_URL = 'http://images.amazon.com/images/P/%s.%02i.LZZZZZZZ.jpg'
-AMAZON_INDICES = (1, 2)
+class ArtSource(object):
+    def __init__(self, log):
+        self._log = log
 
+    def get(self, album):
+        raise NotImplementedError()
 
```

Comment:
It's because those functions don't really *need* a logger, they just use it, so IMHO it makes more sense for the logger to be part of their state. Moreover it shows their common structure.